bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/EntryLocationIndex.java

```diff
@@ -189,6 +190,21 @@ public void delete(long ledgerId) throws IOException {
         deletedLedgers.add(ledgerId);
     }
 
+    public void compact() {
+        try {
+            isCompact = true;
+            locationsDb.compact();
+        } catch (IOException e) {
```

Comment:
Maybe we can remove the catch here due to we have caught the exception outside.

Reply:
addressed